velox/exec/fuzzer/JoinFuzzer.cpp

```diff
@@ -1011,10 +1041,43 @@ void JoinFuzzer::verify(core::JoinType joinType) {
   const bool nullAware =
       isNullAwareSupported(joinType) && vectorFuzzer_.coinToss(0.5);
 
-  const auto numKeys = nullAware ? 1 : randInt(1, 5);
+  // Add boolean/integer join filter 10% of the time.
+  const bool withFilter = vectorFuzzer_.coinToss(0.1);
+  // Null-aware joins allow only one join key.
+  const int numKeys = nullAware ? (withFilter ? 0 : 1) : randInt(1, 5);
+  std::vector<TypePtr> keyTypes = generateJoinKeyTypes(numKeys);
+  std::string filter;
+
+  if (withFilter) {
+    if (vectorFuzzer_.coinToss(0.5)) {
+      keyTypes.push_back(BOOLEAN());
+      filter = vectorFuzzer_.coinToss(0.5)
+          ? fmt::format("t{} = true", keyTypes.size() - 1)
+          : fmt::format("u{} = true", keyTypes.size() - 1);
+    } else {
+      keyTypes.push_back(INTEGER());
+      filter = vectorFuzzer_.coinToss(0.5)
+          ? fmt::format("t{} % {} = 0", keyTypes.size() - 1, randInt(1, 9))
+          : fmt::format("u{} % {} = 0", keyTypes.size() - 1, randInt(1, 9));
+    }
+  }
 
-  // Pick number and types of join keys.
-  const std::vector<TypePtr> keyTypes = generateJoinKeyTypes(numKeys);
+  if (vectorFuzzer_.coinToss(0.1)) {
+    if (nullAware) {
+      keyTypes.clear();
+    }
+    if (vectorFuzzer_.coinToss(0.5)) {
+      keyTypes.push_back(BOOLEAN());
+      filter = vectorFuzzer_.coinToss(0.5)
+          ? fmt::format("t{} = true", keyTypes.size() - 1)
+          : fmt::format("u{} = true", keyTypes.size() - 1);
+    } else {
+      keyTypes.push_back(INTEGER());
+      filter = vectorFuzzer_.coinToss(0.5)
+          ? fmt::format("t{} % {} = 0", keyTypes.size() - 1, randInt(1, 9))
+          : fmt::format("u{} % {} = 0", keyTypes.size() - 1, randInt(1, 9));
+    }
+  }
```

Comment:
nit: This piece of code is no longer needed?

Reply:
Removed.